### ISSUE TITLE
Btop theme disappears (catpuccin_macchiato)

### ISSUE THREAD

**Issue by RedBearAK** [JaKooLit/Fedora-Hyprland#15]
I just noticed that when looking at the `btop` options, the default theme you've configured (catpuccin_macchiato) appears under a header that says it is theme 33 out of 32. And when moving with the arrow keys to pick another theme, there is apparently no way to get back to that theme because it will then loop around from 32 to 1, and from 1 to 32 when going backward. 

So essentially the theme you've pre-configured "disappears" and can no longer be chosen once something else is selected. 

Must be something slightly off about the way the theme is installed, that causes `btop` not to count it as part of the total number of themes. 

Minor issue, obviously. I can't look deeper into it right now to help you with a fix. Just thought I'd let you know.

**Comment by JaKooLit** [comment]
can you check contents of ~/.config/btop/themes?

if themes are present, can you edit ~/.config/btop/btop.conf

and change the theme to $HOME/.config/btop/themes/catppuccin_macchiato.theme

I didnt experienced it so probably stating the whole path will fix

**Comment by RedBearAK** [comment]
That line in the config changes depending on what you pick in the options screen in btop. I'm sure I could manually put the reference back, but that's not really the point.  The location is exactly what it should be, according to the comments in the conf file, so the question is why the four "catpuccin_*" theme files in that folder are not being picked up by btop. 

This is true for running the command in a terminal, and also for running the "btop++" item from the pop-up menu. The user themes just don't show up. 

I think I'll file a bug report on the btop repo, or look for some issues about themes.

**Comment by JaKooLit** [comment]
Oh I understand now what you mean....

Anyway, re tried again, so here's what you should do so it wont be removed so the additional themes that I have added wont be remove.

launch btop through tty or right click on memory.

Change theme... press esc to get out of the menu. and quit

Once you relaunch, the additional catppuccin themes set will be loaded

**Comment by RedBearAK** [comment]
I copied the theme files into /usr/share/btop/themes and they were picked up right away the next time I started btop. It said 36 themes instead of 32. Deleting them from that location makes them no longer available in btop, even though the user theme files are still there. 

I don't understand your instructions, especially the "right click on memory". I've been changing the themes a lot. This just seems like a bug in btop that keeps it from seeing the user themes.

**Comment by JaKooLit** [comment]
> I copied the theme files into /usr/share/btop/themes and they were picked up right away the next time I started btop. It said 36 themes instead of 32. Deleting them from that location makes them no longer available in btop, even though the user theme files are still there.
> 
> I don't understand your instructions, especially the "right click on memory". I've been changing the themes a lot. This just seems like a bug in btop that keeps it from seeing the user themes.

sorry, right click on memory module..... 

I dont really want to copy or mess up users root directory unless it is really necessary.

Again, if you follow my instruction, the themes will load.... DO NOT close or quit btop by super q.... quit the btop from btop menu once you change the themes......

**Comment by RedBearAK** [comment]
> I dont really want to copy or mess up users root directory unless it is really necessary.

Yes, I was just trying to verify there was thing wrong with the theme files themselves. The permissions on the files and the themes folder all seem fine. 

> right click on memory module.....

OK, I saw that it opened btop after right-clicking the memory module on the waybar. 

And now that I've checked a few things and gone through the whole list of themes, I sort of understand what's going on. I was confused by btop initially saying the macchiato theme was "33/32" and then the "33" disappeared when I moved to the right. But there are only 26 default themes in /usr/share/btop/themes. So the catpuccin themes are actually part of the list still, along with apparently two "built-in" themes. They are ordered like this in the list, which helped confuse me into thinking the catpuccin themes were missing: 

- Default
- TTY
- catpuccin_frappe
- catpuccin_latte
- catpuccin_macchiato
- catpuccin_mocha
- HotPurpleTrafficLight
- adapta
- ayu
- dracula
- ... (the rest of the themes that come with btop)

So the "user" themes are inserted into the beginning of the list, but after "Default" and "TTY", and not alphabetically inserted in the list overall (but also capital letters apparently are sorted first in the rest of the list). 

I was looking for the catpuccin themes in the "gap" between theme 32 and theme 1, and just thought they were gone because of the way that "33" disappeared. 

So, this is just sort of a counting bug in btop, the way it counted the themes before I first changed the theme. Something caused it to have an extra number, but that macchiato theme should have been theme number.. 5? Out of 32. 

Weird bug.

**Comment by RedBearAK** [comment]
Seems to have something to do with setting the theme manually in the conf file. I did that and once again saw the "33/32" in the btop options screen, which once again disappeared when I moved to a different theme. Had to move multiple times to the right to get back to the "catpuccin_*" themes, as expected.
